internal/store/manifest_target.go

```diff
@@ -47,6 +47,16 @@ func (mt *ManifestTarget) UpdateStatus() v1alpha1.UpdateStatus {
 		if us == v1alpha1.UpdateStatusInProgress {
 			return v1alpha1.UpdateStatusPending
 		}
+
+		// If the local_resource has auto_init=False and has not built yet (i.e. it has
+		// not been manually triggered), use UpdateStatusNone to indicate that the resource
+		// has not yet had a reason to trigger so that it blocks the serve_cmd from executing.
+		// Once manually triggered, a no-op build will exist, and subsequent calls will return
+		// UpdateStatusNotApplicable so that the server controller knows it does not need to
+		// wait for anything.
+		if !m.TriggerMode.AutoInitial() && !mt.State.StartedFirstBuild() {
```

Comment:
Hmmmm...i'm not sure this is right. a couple obvious smells that leap out to me:

- Currently mt.State.UpdateStatus() does this same check. why does this check live in both places?
- Other things read UpdateStatus too (like the UI), will they also do the right thing in this case?
- I'm also not sure I can articulate the difference between "UpdateStatusNone" and "UpdateStatusNotApplicable". can you?

Reply:
a check that might make more sense to me here is

```
	if us == v1alpha1.UpdateStatusNone {
		return us
	}
```

which is closer to the check above

Reply:
Okay, I simplified the logic so that it only explicitly handles the special cases -- there's _technically_ slight semantic change here now which is that `UpdateStatusError` won't get coerced to `UpdateStatusNotApplicable` anymore, but I think that's a good thing because that'd be indicative of an engine bug (i.e. no-op build "failing" should be impossible) and would probably result in UI being in a confusing/inconsistent state.

RE: The approach, I agree it's odd. IMO while it is kludgy it's at least consistent vs a non-existent update being allowed to be `Pending` but not `None`, i.e. it should either _always_ be `NotApplicable` or it should generally follow the rules. (FWIW I'm also a bit confused why it coerces `InProgress` -> `Pending` instead of making `InProgress` a "hold" status in server controller, but that's a separate issue.)

Alternatively, we can add special logic to server controller for N/A statuses to do a `!m.TriggerMode.AutoInitial() && !mt.State.StartedFirstBuild()` check?

For difference between `None` / `NotApplicable` -- I agree, I think the name is unfortunate but the docs/usage are consistent since it's what gets sued for anything that's `auto_init=False`: https://github.com/tilt-dev/tilt/blob/87a7f57d91aa0685b076e80eb54cfcd63bf0e83f/pkg/apis/core/v1alpha1/updatestatus_types.go#L8-L10

Thankfully the UI does handle it appropriately (also confirmed by actually running): https://github.com/tilt-dev/tilt/blob/87a7f57d91aa0685b076e80eb54cfcd63bf0e83f/web/src/status.tsx#L12-L15